src/other/index.jl

```diff
@@ -158,7 +158,7 @@ function Base.getindex(x::AbstractIndex, idx::AbstractVector)
     end
     length(idxs) == 0 && return Int[] # special case of empty idxs
     if idxs[1] isa Real
-        if !all(v -> v isa Integer && !(v isa Bool), idxs)
```

Comment:
Why change this?

Reply:
because it was breaking when a `Symbol` was passed in

Reply:
Nothing, it was breaking in 0.6 also.  Since it only produced a warning nobody seems to have noticed.

Reply:
@bkamins Does that look OK to you?

Reply:
It was breaking because we do not allow mixing indexing by number and symbol in the same index. It has to be either all numbers or all symbols. We have to exclude `Bool` because we do not allow indexing by `Bool` in this case and `Bool <: Integer`.

Now conversion to `Int` will fail on `Symbol` anyway in line 169.

Reply:
oh, I think I was *very* confused about what the intent was there.  will change.